dashboard/app/models/script.rb

```diff
@@ -1475,10 +1475,28 @@ def pilot?
   end
 
   def has_pilot_access?(user = nil)
-    pilot? && !!user&.permission?(UserPermission::LEVELBUILDER)
+    return false unless pilot? && user
+    return true if user.permission?(UserPermission::LEVELBUILDER)
+    return true if has_pilot_experiment?(user)
+
+    # A user without the experiment has pilot script access if
+    # (1) they have been assigned to or have progress in the pilot script, and
+    # (2) one of their teachers has the pilot experiment enabled.
+    has_progress = !!UserScript.find_by(user: user, script: self)
+    has_progress && user.teachers.any? {|t| has_pilot_experiment?(t)}
```

Comment:
Are we sure this last clause is one we want? I remember some concern that some of our piloters have had many students, and/or may also be facilitators that have had other teachers as students, at some point, granting them descendant access to the script.

Reply:
In order to have progress, the student would have to have been assigned to the script. So, if pilot teacher has sections A, B, C and assigns only C to the pilot script, sections A and B will not gain access. Does that address your concerns?

Reply:
Yep!  Sorry, I didn't parse this correctly the first time.